From 9ae234968078366877a1f8027423644ae7bbfb89 Mon Sep 17 00:00:00 2001
From: Roger Zanoni <rzanoni@igalia.com>
Date: Mon, 13 Nov 2023 11:07:15 -0300
Subject: [PATCH 32/33] [M118-fix] zlib: Fix arm build

Prevent inclusion of cpuid on arm builds.

error:

third_party/zlib/cpu_features.c:15:
cpuid.h:14:2: error: this header is for x86 only

Upstream-Status: Inappropriate
Signed-off-by: Roger Zanoni <rzanoni@igalia.com>
---
 third_party/zlib/adler32_simd.c | 2 +-
 third_party/zlib/cpu_features.c | 2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/third_party/zlib/adler32_simd.c b/third_party/zlib/adler32_simd.c
index 58966eecf0b80..25867ace82bd2 100644
--- a/third_party/zlib/adler32_simd.c
+++ b/third_party/zlib/adler32_simd.c
@@ -49,7 +49,7 @@
 /* NMAX is the largest n such that 255n(n+1)/2 + (n+1)(BASE-1) <= 2^32-1 */
 #define NMAX 5552
 
-#if defined(ADLER32_SIMD_SSSE3)
+#if (__x86_64__ || __i386__) && defined(ADLER32_SIMD_SSSE3)
 
 #include <tmmintrin.h>
 
diff --git a/third_party/zlib/cpu_features.c b/third_party/zlib/cpu_features.c
index 64e0428cd2fc2..c26c619970c10 100644
--- a/third_party/zlib/cpu_features.c
+++ b/third_party/zlib/cpu_features.c
@@ -11,7 +11,7 @@
 #include <stdint.h>
 #if defined(_MSC_VER)
 #include <intrin.h>
-#elif defined(ADLER32_SIMD_SSSE3)
+#elif (__x86_64__ || __i386__) &&  defined(ADLER32_SIMD_SSSE3)
 #include <cpuid.h>
 #endif
 
-- 
2.42.1

